Robinhood/Robinhood.py

```diff
@@ -759,15 +861,51 @@ def market_value(self):
 
         return float(self.portfolios()['market_value'])
 
+    def instrument_url(self,instrument=None):
+        """
 
-    def order_history(self):
-        """Wrapper for portfolios
+        return the canonical instrument URL given either an instrument dict object or string
+        that represents and instrument
 
-            Returns:
-                (:obj:`dict`): JSON dict from getting orders
+        Args:
+            instrument (string): RH instrument URL, instrumentid or instrument object
         """
+        url = None
+        if isinstance(instrument, dict) and 'url' in instrument:
```

Comment:
again with the if/then tree.  This is a huge bear to test and should be able to be made much simpler